### PR TITLE
[FIX]: Image occlusion cards created on desktop app now fit to width 

### LINKED ISSUE #15235: [BUG]: Images in Image Occlusion cards are too small on AnkiDroid [landscape mode]
### Checked for duplicates?

- [X] This issue is not a duplicate

### What are the steps to reproduce this bug?

Create an Image Occlusion card in Anki desktop (Version ⁨23.12.1).
Sync and review it in AnkiDroid.

### Expected behaviour

Images fit to width.

### Actual behaviour

Images are very small and have to be zoomed in manually.

This is a card made with Anki's native Image Occlusion:

test1

This was made using the Image Occlusion Enhanced addon:

test2

### Debug info

```text
AnkiDroid Version = 2.17alpha15 (e5b25f3fef737140ae396d1b212a8d638551500f)

Backend Version = 0.1.34-anki23.12.1 (23.12.1 1a1d4d5419c6b57ef3baf99c9d2d9cf85d36ae0a)

Android Version = 12 (SDK 31)

ProductFlavor = full

Manufacturer = HMD Global

Model = Nokia 5.4

Hardware = qcom

Webview User Agent = Mozilla/5.0 (Linux; Android 12; Nokia 5.4 Build/SKQ1.220119.001; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/120.0.6099.193 Mobile Safari/537.36

ACRA UUID = e3cee8ce-3afa-4813-a03d-d23e62bc8a76

Crash Reports Enabled = false
```

### (Optional) Anything else you want to share?

Update: the issue is still present in AnkiDroid 2.17alpha16

### Research

- [X] I am reporting a bug specific to AnkiDroid (Android app)
- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [X] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by RedLexa**
## Purpose / Description
This commit applies the suggested solution in #15235, ensuring Image Occlusion cards created in the desktop app fit to width. 

## Fixes
* Fixes  #15235

## Approach
Applying the correct styling with !important to make sure it overwrites whatver came over from the Desktop app that was causing issues.

## How Has This Been Tested?

I ran this with multiple image sizes, creating them in the desktop app and then opening then in AnkiDroid while in landscape mode.

## Checklist

- [X] You have a descriptive commit message with a short title (first line, max 50 chars).
- [X] You have commented your code, particularly in hard-to-understand areas
- [X] You have performed a self-review of your own code

### CODE REVIEW

**Comment by david-allison** [comment]
Thanks! Could you provide before/after screenshots in portrait and landscape?

File: AnkiDroid/src/main/assets/flashcard.css [hunk 1]
```diff
@@ -131,3 +131,36 @@ scale up, but not down.  */
...
+  object-fit: contain !important;
+}
+
+.card {
+  font-family: arial;
+  font-size: 20px;
+  text-align: center;
+  color: black;
+  background-color: white;
+}
```
Comment (david-allison) [inline AnkiDroid/src/main/assets/flashcard.css:157]: This very likely shouldn't be here, as it should come from the card template

**Comment by RedLexa** [comment]
> Thanks! Could you provide before/after screenshots in portrait and landscape?
> 
> Did you also resolve: [#15235 (comment)](https://github.com/ankidroid/Anki-Android/issues/15235#issuecomment-2004609689)

I did resolve https://github.com/ankidroid/Anki-Android/issues/15235#issuecomment-2004609689 ! I wasn't putting !important on the right properties, that fixed it.
As for the screenshots, here they are:
BEFORE
Hacksaw RIdge normal
Hacksaw RIdge em Landscape
AFTER
Hacksaw final  Landscape
Hacksaw final normal

File: AnkiDroid/src/main/assets/flashcard.css (hunk 1 above)
Comment (RedLexa) [reply AnkiDroid/src/main/assets/flashcard.css:157]: Will remove, test again just to make sure and commit again.

**Comment by david-allison** [comment]
And to confirm, that's the intended outcome (as all the text isn't ocluded)

**Comment by RedLexa** [comment]
> And to confirm, that's the intended outcome (as all the text isn't occluded)

Yes! That was just a placeholder occlusion I placed to test if it was working.
(You aren't to first to point out how slightly annoying it is that the occlusion just barely covers all the text, I'll try to do better next time :) )

File: AnkiDroid/src/main/assets/flashcard.css (hunk 1 above)
Comment (RedLexa) [reply AnkiDroid/src/main/assets/flashcard.css]: Done!

**Review by david-allison — APPROVED** [review]
Fantastic, thanks so much for doing the deep dive and seeing this through!

File: AnkiDroid/src/main/assets/flashcard.css [hunk 2]
```diff
@@ -131,3 +131,28 @@ scale up, but not down.  */
...
   fill: white;
 }
+
+/* Make sure the image occlusion cards fit to width in
+landscape mode, even if made in Desktop version*/
+#image-occlusion-canvas {
+  --inactive-shape-color: #ffeba2;
+  --active-shape-color: #ff8e8e;
+  --inactive-shape-border: 1px #212121;
+  --active-shape-border: 1px #212121;
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/assets/flashcard.css:141]: what is the relation of these colors with the occlusions size? Are they necessary? If so, why?

Comment (krmanik) [reply AnkiDroid/src/main/assets/flashcard.css:141]: We don't need these changes, these are available in notetype of image occlusion.

**Comment by krmanik** [comment]
The backside of card hides toggle button,

</img>

**Comment by RedLexa** [comment]
> The backside of card hides toggle button,
> 
> 

I'm sorry, I d

**Comment by RedLexa** [comment]
Hey everyone. Sorry for the late reply, school got in the way. I'm afraid I have some bad news: the new UI for the card reviewing seems to have messed with something that was making the solution work beforehand. ( https://github.com/ankidroid/Anki-Android/pull/16109#issuecomment-2040596362) Now, the issue seems to be back:
 
image

However, through my testing, even cards Image Occlusion cards seem created directly on AnkiDroid seem to be having this issue. I'm stumped on what to do here. What changes that were implemented in the last 3 weeks that might mess with the card css? Thank you all again for your patience and sorry for the delay.

**Comment by david-allison** [comment]
@RedLexa are you testing on `main`? We recently applied a fix for some JS errors in the Reviewer

**Comment by david-allison** [comment]
You need to manually rebase it onto the current `main`

**Comment by BrayanDSO** [comment]
FWIW, this is the CSS that I ~~stole~~ retrieved from inspecting AnkiMobile:

```css
#image-occlusion-container {
    position: relative;
    margin: 0 auto;
    max-height: calc(100vw - 24px) // this was `100vw - var(--io-header)`, which defaults to 24px but can change I think. I don't think that we want to implement something similar
}

#image-occlusion-container img {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    max-width: unset;
    max-height: unset
}
```

**Comment by RedLexa** [comment]
Testing after having rebased had the same result, unfortunately.

@BrayanDSO Where did you get this styling? Was it from inspecting the image occlusion card's styling? The styling I'm applying should overwrite this and fix it, for some reason it isn't.

**Pull request closed without merging**